plugin/storage/cassandra/schema/create.sh

```diff
@@ -42,6 +42,9 @@ fi
 
 keyspace=${KEYSPACE:-"jaeger_v1_${datacenter}"}
 
+#replaces hyphen and period with underscore
+keyspace=$(echo $keyspace | sed 's/[-.]/_/g')
```

Comment:
I don't think this is the best fix for the original problem. For example, I can still give a weird datacenter name like `dc#1` and the CQL will blow up on invalid keyspace name.  A better solution is to simply check for invalid characters (via exclusion, i.e. `[^a-zA-z0-9_]`) and throw an error, advising the user to make their own decision and pass the keyspace as KEYSPACE env var.